ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -265,22 +275,51 @@ private static SqlDropConnectorStatement getDropConnectorStatement(final String
     return new SqlDropConnectorStatement(sql, dropConnectorMatcher.group(1));
   }
 
-  private static SqlPropertyCommand getSetPropertyCommand(final String sql) {
+  private static SqlPropertyCommand getSetPropertyCommand(
+      final String sql, final Map<String, String> variables) {
     final Matcher setPropertyMatcher = SET_PROPERTY.matcher(sql);
     if (!setPropertyMatcher.matches()) {
       throw new MigrationException("Invalid SET command: " + sql);
     }
     return new SqlPropertyCommand(
-        sql, true, setPropertyMatcher.group(1), Optional.of(setPropertyMatcher.group(2)));
+        sql,
+        true,
+        VariableSubstitutor.substitute(setPropertyMatcher.group(1), variables),
+        Optional.of(VariableSubstitutor.substitute(setPropertyMatcher.group(2), variables))
+    );
   }
 
-  private static SqlPropertyCommand getUnsetPropertyCommand(final String sql) {
+  private static SqlPropertyCommand getUnsetPropertyCommand(
+      final String sql, final Map<String, String> variables) {
     final Matcher unsetPropertyMatcher = UNSET_PROPERTY.matcher(sql);
     if (!unsetPropertyMatcher.matches()) {
       throw new MigrationException("Invalid UNSET command: " + sql);
     }
     return new SqlPropertyCommand(
-        sql, false, unsetPropertyMatcher.group(1), Optional.empty());
+        sql,
+        false,
+        VariableSubstitutor.substitute(unsetPropertyMatcher.group(1), variables),
+        Optional.empty());
+  }
+
+  private static SqlDefineVariableCommand getDefineVariableCommand(final String sql) {
+    final Matcher defineVariableMatcher = DEFINE_VARIABLE.matcher(sql);
```

Comment:
Haven't tried this so I might be mistaken but I think we've got a bug around casing here: `transformToSqlCommand()` uppercases the sql before applying the regex matchers but we don't do that here (we can't because we don't want to strings or quoted identifiers), which means this matcher might fail to find a match (e.g., if the user lowercases `DEFINE` in their sql command). I think this bug also affects the other statement types. 

If this is true, we definitely need to fix this, and we should add test coverage for it too.

Reply:
Added a fix

Reply:
Thanks. Do you know whether this is a problem for 0.17 (e.g., for INSERT VALUES)? If so, we should backport a fix to 6.2.x (requires going through the CP blocker process). We should also consider getting a fix into 0.17 but I think that's unlikely at this point, in which case we'd want to document the limitation.

Reply:
It affects SET and UNSET. Filed a blocker, but IMO because there is a clear workaround (tell users to capitalize SET and UNSET), it's not *terrible* if it doesn't get included.

Reply:
Agreed it's not terrible if it's not included but I think it's still better to fix it before the 6.2 release. 0.17 probably isn't worth it, though.